### PR TITLE
Refactor ViewSource as per sugarlabs/sugar-toolkit-gtk3#313

### PR DESCRIPTION
**Pull request by AbrahmAB**
An example of implementation of pop-up class made in sugar3.graphics
https://github.com/sugarlabs/sugar-toolkit-gtk3/pull/313

### CODE REVIEW

**Comment by samdroid-apps** [comment]
That code looks nicer that it did before your patch!  Good job!

**Comment by AbrahmAB** [comment]
Thank you :)

**Comment by quozl** [comment]
Please add more explanation to the commit description and force push.

**Comment by AbrahmAB** [comment]
@quozl Changed commit message.

**Comment by quozl** [comment]
Thanks, but the commit message should not refer to a GitHub pull request, but a git commit hash.  So that git repository alone contains all historical data.

Citing a git commit is best done with the hash and the first line of the commit message.  Since this would be a cross-repository citing, also add repository name, for example;

* sugar-toolkit-gtk3:fc3b08a ("Added new popwindow module in sugar3.graphics")

Here are two citations I've had to prepare recently on other work:

* kernel:82ef33a ("Staging: olpc_dcon: Remove obsolete driver")
* kernel:53c43c5 ('Revert "Staging: olpc_dcon: Remove obsolete driver"')
* kernel:3cbaa59 ("perf: Fix ctx time tracking by introducing EVENT_TIME")

**Comment by AbrahmAB** [comment]
simply using "sugar-toolkit-gtk3:fc3b08a" would be enough?
because the commit message `Refactoring ViewSource as per sugar-toolkit-gtk3:fc3b08a ("Added new popwindow module in sugar3.graphics")` would be too long!

**Comment by quozl** [comment]
Use line breaks; the first line is summary of the commit, then a blank line, then details.  You would put the citation in the details.

Look at previous commits in the repository to get an idea of how long they can be.  Unlimited.  We encourage a full explanation of the commit in the message; what the problem is, how it was solved, what other solutions were rejected and why, and a list of references.

**Comment by AbrahmAB** [comment]
@quozl commit details added.

File: src/jarabe/view/viewhelp.py [hunk 1]
```diff
@@ -150,27 +151,12 @@ def __init__(self, activity, window_xid):
...
-        self.connect('realize', self.__realize_cb)
-        self.connect('hide', self.__hide_cb)
-        self.connect('key-press-event', self.__key_press_event_cb)
+        PopWindow.__init__(self, title, set_titlebox=False,
+                           window_xid=window_xid)
+        self._parent_window_xid = window_xid
 
         toolbar = Toolbar(title, has_local_help)
-        box.pack_start(toolbar, False, False, 0)
+        self._vbox.pack_start(toolbar, False, False, 0)
```
Comment (samdroid-apps) [inline src/jarabe/view/viewhelp.py]: can we refactor it to use the toolbar api provided by the PopWindow?

Comment (AbrahmAB) [inline src/jarabe/view/viewhelp.py]: Done! :)

**Comment by samdroid-apps** [comment]
When applying this patch and the latest sugar3 patches, I get an error when opening view help or view source:

```
Traceback (most recent call last):
  File "/home/saam/sugar-build/build/out/install/lib/python2.7/site-packages/jarabe/view/palettes.py", line 146, in __view_source__cb
    setup_view_source(self._home_activity)
  File "/home/saam/sugar-build/build/out/install/lib/python2.7/site-packages/jarabe/view/viewsource.py", line 163, in setup_view_source
    sugar_toolkit_path, activity.get_title())
  File "/home/saam/sugar-build/build/out/install/lib/python2.7/site-packages/jarabe/view/viewsource.py", line 175, in __init__
    window_xid=window_xid)
  File "/home/saam/sugar-build/build/out/install/lib/python2.7/site-packages/sugar3/graphics/popwindow.py", line 61, in __init__
    Gtk.Window.__init__(self, **kwargs)
  File "/usr/lib64/python2.7/site-packages/gi/overrides/__init__.py", line 326, in new_init
    return super_init_func(self, **new_kwargs)
TypeError: gobject `SugarViewSource' doesn't support property `set_titlebox'

Traceback (most recent call last):
  File "/home/saam/sugar-build/build/out/install/lib/python2.7/site-packages/jarabe/view/palettes.py", line 154, in __view_help__cb
    setup_view_help(self._home_activity)
  File "/home/saam/sugar-build/build/out/install/lib/python2.7/site-packages/jarabe/view/viewhelp.py", line 134, in setup_view_help
    viewhelp = ViewHelp(activity, window_xid)
  File "/home/saam/sugar-build/build/out/install/lib/python2.7/site-packages/jarabe/view/viewhelp.py", line 151, in __init__
    window_xid=window_xid)
  File "/home/saam/sugar-build/build/out/install/lib/python2.7/site-packages/sugar3/graphics/popwindow.py", line 61, in __init__
    Gtk.Window.__init__(self, **kwargs)
  File "/usr/lib64/python2.7/site-packages/gi/overrides/__init__.py", line 326, in new_init
    return super_init_func(self, **new_kwargs)
TypeError: gobject `jarabe+view+viewhelp+ViewHelp' doesn't support property `set_titlebox'
```

**Comment by AbrahmAB** [comment]
Yes, I need to update this patch and make it compatible with the new changes in sugar3. I'll do it asap.

**Comment by AbrahmAB** [comment]
Error fixed!

**Comment by AbrahmAB** [comment]
Changes made as per the new commit in [#313](https://github.com/sugarlabs/sugar-toolkit-gtk3/pull/313)

**Comment by samdroid-apps** [comment]
I merged the sugar3 patch, but not this one.  Just a few little issues:

* Regression; no longer the padding on the left side icons in view source
* Regression; left side icons have reversed their order in both the view source and view help

I also found a bug, that seems to be very odd.  Steps to reproduce:

1.  Open view help for an activity
2.  Open frame
3.  Right click that activity's icon in the frame
4.  Mouse onto the palette
5.  Expected behaviour; mouse events are sent to the palette, you can click on the stop menu item
6.  Actual behaviour:  mouse events are not sent to the palette.  The palette goes away after 1sec, even if you keep your mouse inside it.  Clicking the stop menu item does nothing.

So good job making the popwindow's steal focus from other windows... but they shouldn't steal focus from the frame's palettes :)

**Pull request closed without merging**